src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -0,0 +1,289 @@
+    }
+
+    finalize(): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        if (this._adyenOldComponent) {
+            this._adyenOldComponent.unmount();
+            this._adyenOldComponent = undefined;
+        } else if (this._adyenComponent) {
+            this._adyenComponent.unmount();
+            this._adyenComponent = undefined;
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _getAdyenV2PaymentInitializeOptions(): AdyenV2PaymentInitializeOptions {
+        if (!this._adyenv2) {
+            throw new InvalidArgumentError(' "options.adyenv2" argument was not provided during initialization.');
+        }
+
+        return this._adyenv2;
+    }
+
+    private _getLocale(): string {
+        const state = this._store.getState();
+        const storeConfig = state.config.getStoreConfig();
+
+        if (!storeConfig) {
+            return 'en_US';
+        }
+
+        return storeConfig.storeProfile.storeLanguage;
+    }
+
+    private _getStateContainer(): string {
+        if (!this._stateContainer) {
+            return '{}';
+        }
+
+        return this._stateContainer;
+    }
+
+    private _getThreeDS2ChallengeWidgetSize(): string {
+        const { threeDS2ChallengeWidgetSize } = this._getAdyenV2PaymentInitializeOptions();
+
+        if (!threeDS2ChallengeWidgetSize) {
+            return '01';
+        }
+
+        return threeDS2ChallengeWidgetSize;
+    }
+
+    private _handle3DS2Challenge(resultObject: ThreeDS2Result, paymentMethodId: string): Promise<Payment> {
+
+        return new Promise((resolve, reject) => {
+            if (!this._adyenCheckout) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            const threeDS2Component = this._adyenCheckout
+                .create(ThreeDS2ComponentType.ThreeDS2Challenge, {
+                    challengeToken: resultObject.token,
+                    onComplete: (challengeData: ThreeDS2OnComplete) => {
+                        const challengePaymentPayload = {
+                            ...challengeData.data,
+                            paymentData: resultObject.payment_data,
+                        };
+
+                        const paymentPayload = {
+                            methodId: paymentMethodId,
+                            paymentData: {
+                                nonce: JSON.stringify(challengePaymentPayload, null, 2),
+                            },
+                        };
+
+                        resolve(paymentPayload);
+                    },
+                    onError: (error: AdyenError) => reject(error),
+                    size: this._getThreeDS2ChallengeWidgetSize(),
+                });
+
+            threeDS2Component.mount(`#${this._getAdyenV2PaymentInitializeOptions().containerId}-3ds`);
```

Comment:
I think it's fine to provide a default value of the container. But you should also allow the client app to pass in a container ID for the 3DS component. The reason for this is because the client app wouldn't know they need to have a container with the `-3ds` suffix. If you make this an option, it'd be documented and the client app would know they need to have such container in the DOM in advance.

Reply:
Yes actually this is by Adyen requirement, to mount a new component in the same page (now is handled by a modal (I already have this in this branch => https://github.com/clopezh/checkout-sdk-js/tree/INT-1901) so in this case which is the process to document it ?

Reply:
Do we just need to add a comment in this line?

Reply:
What I meant is to have both `containerId` (which refers to the default credit card component) and `threeDS2ContainerId`. And if `threeDS2ContainerId` is not provided, you can fallback to `${containerId}-3ds`.

Reply:
Correct I'll added